Makefile

```diff
@@ -1,30 +1,65 @@
+# Only use the recipes defined in these makefiles
 MAKEFLAGS += --no-builtin-rules
-.PHONY: ${binary} dc gen test
 .SUFFIXES:
+# Delete target files if there's an error
+# This avoids a failure to then skip building on next run if the output is created by shell redirection for example
+# Not really necessary for now, but just good to have already if it becomes necessary later.
+.DELETE_ON_ERROR:
+# Treat the whole recipe as a one shell script/invocation instead of one-per-line 
+.ONESHELL:
+# Use bash instead of plain sh 
+SHELL := bash
+.SHELLFLAGS := -o pipefail -euc
 
 binary := boots
+.PHONY: all ${binary} crosscompile dc gen run test
 all: ${binary}
 
-crosscompile: $(shell git ls-files | grep -v -e vendor -e '_test.go' | grep '.go$$' )
-	CGO_ENABLED=0 GOOS=linux GOARCH=386 go build -v -o ./boots-linux-x86_64 -ldflags="-X main.GitRev=$(shell git rev-parse --short HEAD)"
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -v -o ./boots-linux-amd64 -ldflags="-X main.GitRev=$(shell git rev-parse --short HEAD)"
-	CGO_ENABLED=0 GOOS=linux GOARCH=arm GOARM=6 go build -v -o ./boots-linux-aarch64 -ldflags="-X main.GitRev=$(shell git rev-parse --short HEAD)"
-	CGO_ENABLED=0 GOOS=linux GOARCH=arm GOARM=7 go build -v -o ./boots-linux-armv7l -ldflags="-X main.GitRev=$(shell git rev-parse --short HEAD)"
-	CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go build -v -o ./boots-linux-arm64 -ldflags="-X main.GitRev=$(shell git rev-parse --short HEAD)"
-
+GitRev := $(shell git rev-parse --short HEAD)
+crosscompile: ${binary}
```

Comment:
I have to validate how this will break the release process in sandbox. It will do it for sure, and we should probably document and align the other project and binaries as well otherwise it will be a pretty messy experience.

Reply:
This won't change any of that, at least that was the intent.

Reply:
ohhh thats a mistake, not meant for this PR! Let me drop that!

Reply:
oh you are right! Git diff tricked me out!

Reply:
ok yes I got tricked too :D

Reply:
Nothing has changed here :D